frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -73,6 +67,38 @@ export default function SurveyEdit(): JSX.Element {
         setSelectedPageIndex(newIndex)
     }
 
+    const initialThankYouMessageDescriptionContentType = surveysHTMLAvailable
+        ? thankYouMessageDescriptionContentType || 'text'
+        : 'text'
+
+    const handleSurveyValueChange = (key: string, val: string): void => {
+        const updatedAppearance = {
+            ...survey.appearance,
+            [key]: val,
+        }
+        setSurveyValue('appearance', updatedAppearance)
+    }
+
+    const updateWritingHTMLDescription = (val: boolean): void => {
+        if (surveysHTMLAvailable) {
+            handleSurveyValueChange('thankYouMessageDescriptionContentType', val ? 'html' : 'text')
+        }
+    }
+
+    const onTabChange = (key: string): void => {
+        if (surveysHTMLAvailable || key === 'text') {
+            updateWritingHTMLDescription(key === 'html')
+        }
+    }
+
+    const handleChange = (val: string): void => {
+        setSurveyValue('appearance', {
+            ...survey.appearance,
+            thankYouMessageDescription: val,
+            thankYouMessageDescriptionContentType,
+        })
+    }
```

Comment:
This method applies only to the `HTMLEditor` component. I suggest either renaming it to `handleHTMLEditorChange` or (my preference), passing it inline to the component (line 281).

We already do this for other components (e.g., lines 263 and 295). The benefit is that it keeps the `SurveyEdit` scope less cluttered.